concept/type_/mod.rs

```diff
@@ -283,6 +283,22 @@ pub trait OwnerAPI<'a>: TypeAPI<'a> {
         attribute_type: AttributeType<'static>,
     ) -> Result<MaybeOwns<'m, HashSet<CapabilityConstraint<Owns<'static>>>>, ConceptReadError>;
 
+    fn get_owned_attribute_types_declared(
```

Comment:
My only thinking is that this has a memory/performance performance profile from the `get_owns_declared`, since it has to allocate a HashSet each time you call it.
1. Do you need this API?
2. Does it guarantee extra Uniqueness (not getting duplicate AttributeTypes) that is useful to you?

If yes to both - i think it might be worth adding a API for this that doesn't rely on the other `get_owns_declared`, and has to be implemented in each implementation -> can be a new or different cache entry to avoid the allocation...

On the other hand, this is the kind of premature optimisation that led us to some technical debt and probably not worth doing! How do we feel about doing a middle ground of prefixing `compute_xxxx` when it's a Types ConceptAPI that does extra allocations and dropping that prefix when it is not? Does that mess up readability too much?

Reply:
I was hesitant about creating these APIs. The facts that inclined me to doing it:
* We already have similar "helper" apis with clones, and we are forced to do these clones because we have `MaybeOwns`. Even if we don't do it, we have to clone on the API user (in this case `executor`) side
* We actually do clones, collects, and other "greedy" (if applicable here)  operations for these types/capabilities multiple times in executors
* Without these apis, Dmitrii had an impression that `get_owns` returns just all owns a type has, with this type as the owner if it's an inherited owns. I didn't want to change anything globally, it would break many things (and I'm not sure if it's needed), but we agreed that it's not that bad if we at least had an api to answer the question "can I get all the attributes this type owns?". That's the answer
* The overall usage looked cleaner this way


Based on this, I'm not sure if we need to push any concept api interfaces optimization now since we don't have clear numbers, and with the limited resources, we should prioritize. It looks consistent enough in the context of other concept API calls now, although I agree with the drawbacks we have now.

We can always add these caches, it's easy, but a bit annoying work. Let's do it later and globally. I'm not against of creating a list of hypotheses to check our numbers against later!